tests/plugins/filesystem/test_uuid.py

```diff
@@ -0,0 +1,97 @@
+from io import BytesIO
+from types import SimpleNamespace
+from unittest.mock import patch
+from uuid import UUID
+
+from dissect.target.filesystem import Filesystem
+from dissect.target.filesystems.exfat import ExfatFilesystem
+from dissect.target.filesystems.extfs import ExtFilesystem
+from dissect.target.filesystems.fat import FatFilesystem
+from dissect.target.filesystems.ntfs import NtfsFilesystem
+from tests._utils import absolute_path
+
+
+class MockFilesystem(Filesystem):
+    __type__ = "test"
+
+
+dummy_fh = BytesIO(b"")  # empty in-memory file handle
+
+
+def test_filesystem_uuid_from_volume_guid() -> None:
+    """Filesystem.uuid returns a UUID when volume.guid is set."""
+    guid_bytes = b"\x01" * 16
+    volume = SimpleNamespace(guid=guid_bytes)
+
+    fs = MockFilesystem()
+    fs.volume = volume
+
+    assert fs.uuid == UUID(bytes_le=guid_bytes)
```

Comment:
```suggestion
    guid_bytes = b"\x01" * 16

    fs = MockFilesystem()
    fs.volume = Mock(guid=guid_bytes)

    assert fs.uuid == UUID(bytes_le=guid_bytes)
```
Is there a specific reason you use `SimpleNamespace` instead of just using `Mock` ?